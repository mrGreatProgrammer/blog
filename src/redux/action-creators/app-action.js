export const SET_DARK_SIDE = "SET_DARK_SIDE";
export const SET_LIGHT_SIDE = "SET_LIGHT_SIDE";

export const setDarkSideAC = () => ({ type: SET_DARK_SIDE });
export const setLightSideAC = () => ({ type: SET_LIGHT_SIDE });
