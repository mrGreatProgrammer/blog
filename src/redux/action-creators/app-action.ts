import { AppActionsType, SetDarkSideAction, SetLightSideAction } from "../types/app-reducer";

export const setDarkSideAC = ():SetDarkSideAction => ({ type: AppActionsType.SET_DARK_SIDE });
export const setLightSideAC = ():SetLightSideAction => ({ type: AppActionsType.SET_LIGHT_SIDE });
