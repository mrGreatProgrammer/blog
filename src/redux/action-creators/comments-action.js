export const GET_COMMENTS = "GET_COMMENTS";

export const getCommentsAC = (postId, comments) => ({
  type: GET_COMMENTS,
  postId,
  comments,
});
