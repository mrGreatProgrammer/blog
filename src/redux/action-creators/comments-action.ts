import {
  CommentsActionsType,
  GetCommentsAction,
  IComment,
} from "../types/comments-reducer";

export const getCommentsAC = ( postId: number, comments: any[]): GetCommentsAction =>
 ({
  type: CommentsActionsType.GET_COMMENTS,
  postId,
  comments,
});
