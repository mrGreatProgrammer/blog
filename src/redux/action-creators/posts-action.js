export const GET_POSTS = "GET_POSTS";
export const TOGGLE_IS_FETCHING_POSTS = "TOGGLE_IS_FETCHING_POSTS";
export const SET_CURRENT_PAGE = "SET_CURRENT_PAGE";
export const SET_TOTAL_POSTS_COUNT = "SET_TOTAL_POSTS_COUNT";
export const SHOW_MODAL_WINDOW = "SHOW_MODAL_WINDOW";
export const HIDE_MODAL_WINDOW = "HIDE_MODAL_WINDOW";
export const LIKE_POST = "LIKE_POST";
export const DIS_LIKE = "DIS_LIKE";
export const FILTER_POSTS_BY_DATE = "FILTER_POSTS_BY_DATE";
export const FILTER_POSTS_BY_ALPHABET = "FILTER_POSTS_BY_ALPHABET";

// ACTION CREATORS
export const getPostsAC = (posts, photoInfo) => ({
  type: GET_POSTS,
  posts,
  photoInfo,
});
export const toggleIsFetchingAC = (isFetching) => ({
  type: TOGGLE_IS_FETCHING_POSTS,
  isFetching,
});
export const setCurrentPageAC = (currentPage) => ({
  type: SET_CURRENT_PAGE,
  currentPage,
});
export const setTotalPostsCountAC = (totalPostsCount) => ({
  type: SET_TOTAL_POSTS_COUNT,
  totalPostsCount,
});
export const showModalWindowAC = (postId, postData, photo) => ({
  type: SHOW_MODAL_WINDOW,
  postId,
  postData,
  photo,
});
export const hideModalWindowAC = (postId) => ({
  type: HIDE_MODAL_WINDOW,
  postId,
});

export const likePostAC = (postId, likes) => ({
  type: LIKE_POST,
  postId,
  likes: likes+=1,
});
export const disLikePostAC = (postId, likes) => ({
  type: DIS_LIKE,
  postId,
  likes,
});
export const filterByDateAC = (filterByDate) => ({
  type: FILTER_POSTS_BY_DATE,
  filterByDate,
});
export const filterByAlphabetAC = (filterByAlphabet) => ({
  type: FILTER_POSTS_BY_ALPHABET,
  filterByAlphabet,
});
