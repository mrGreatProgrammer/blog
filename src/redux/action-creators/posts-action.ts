import { DisLikeAction, FilterByAlphabetAction, FilterByDateAction, filterType, GetPostsAction, HideModalWindowAction, LikePostAction, PostsActionsType, SetCurrentPageAction, SetTotalPostsCountACtion, ShowModalWindowAction, ToggleIsFetchingAction } from "../types/posts-reducer";


// ACTION CREATORS
export const getPostsAC = (posts: [], photoInfo: []):GetPostsAction => ({
  type: PostsActionsType.GET_POSTS,
  posts,
  photoInfo,
});
export const toggleIsFetchingAC = (isFetching: boolean):ToggleIsFetchingAction => ({
  type: PostsActionsType.TOGGLE_IS_FETCHING_POSTS,
  isFetching,
});
export const setCurrentPageAC = (currentPage: number):SetCurrentPageAction => ({
  type: PostsActionsType.SET_CURRENT_PAGE,
  currentPage,
});
export const setTotalPostsCountAC = (totalPostsCount: number):SetTotalPostsCountACtion => ({
  type: PostsActionsType.SET_TOTAL_POSTS_COUNT,
  totalPostsCount,
});
export const showModalWindowAC = (postId: number, postData: {}, photo: []):ShowModalWindowAction => ({
  type: PostsActionsType.SHOW_MODAL_WINDOW,
  postId,
  postData,
  photo,
});
export const hideModalWindowAC = (postId: number):HideModalWindowAction => ({
  type: PostsActionsType.HIDE_MODAL_WINDOW,
  postId,
});

export const likePostAC = (postId: number, likes: number):LikePostAction => ({
  type: PostsActionsType.LIKE_POST,
  postId,
  likes: likes+=1,
});
export const disLikePostAC = (postId: number, likes: number):DisLikeAction => ({
  type: PostsActionsType.DIS_LIKE,
  postId,
  likes: likes-=1,
});
export const filterByDateAC = (filterByDate: filterType ):FilterByDateAction => ({
  type: PostsActionsType.FILTER_POSTS_BY_DATE,
  filterByDate,
});
export const filterByAlphabetAC = (filterByAlphabet: filterType):FilterByAlphabetAction => ({
  type: PostsActionsType.FILTER_POSTS_BY_ALPHABET,
  filterByAlphabet,
});
