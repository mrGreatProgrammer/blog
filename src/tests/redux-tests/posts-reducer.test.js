import { getPostsAC, likePostAC, setCurrentPageAC, setTotalPostsCountAC, toggleIsFetchingAC } from "../../redux/action-creators/posts-action";
import postsReducer from "../../redux/reducers/posts-reducer";

let postsForTest = [
  {
    userId: 1,
    id: 1,
    title: "first post title",
    textContent: "first post text content",
    liked: false,
    likes: 3,
    isModalVisible: false,
    info: {
      albumId: 1,
      id: 1,
      title: "accusamus beatae ad facilis cum similique qui sunt",
      url: "https://via.placeholder.com/600/92c952",
      thumbnailUrl: "https://via.placeholder.com/150/92c952",
    },
  },
  {
    userId: 3,
    id: 2,
    title: "second post title",
    textContent: "second post text content",
    liked: false,
    likes: 5,
    isModalVisible: false,
    info: {
      albumId: 2,
      id: 2,
      title: "accusamus beatae ad facilis cum similique qui sunt",
      url: "https://via.placeholder.com/600/92c952",
      thumbnailUrl: "https://via.placeholder.com/150/92c952",
    },
  },
  {
    userId: 1,
    id: 3,
    title: "third post title",
    textContent: "third post text content",
    liked: false,
    likes: 0,
    isModalVisible: false,
    info: {
      albumId: 3,
      id: 3,
      title: "accusamus beatae ad facilis cum similique qui sunt",
      url: "https://via.placeholder.com/600/92c952",
      thumbnailUrl: "https://via.placeholder.com/150/92c952",
    },
  },
];

let state = {
  currentPage: 1,
  pageSize: 2,
  posts: [],
  isFetching: false,
  totalPostsCount: 0,
  filterBy: "",
};

describe("---- GET ALL POSTS ----", () => {
  it("posts length should be three", () => {
    let action = getPostsAC(postsForTest);
    let newState = postsReducer(state, action);

    expect(newState.posts.length).toBe(3);
  });
  it("posts values should be correct", () => {
    let action = getPostsAC(postsForTest);
    let newState = postsReducer(state, action);

    expect(newState.posts).toMatchSnapshot();
  });
});

describe("---- TOGGLE IS FETCHING -----", ()=>{
    it("isFetching should be true", ()=>{
        let action = toggleIsFetchingAC(true);
        let newState = postsReducer(state, action);

        expect(newState.isFetching).toBe(true)
    })
    it("isFetching should be false", ()=>{
        let action = toggleIsFetchingAC(false);
        let newState = postsReducer(state, action);

        expect(newState.isFetching).toBe(false)
    })
})

describe("----- SET CURRENT PAGE -----", ()=>{
    it("currentPage should be one", ()=>{
        let action = setCurrentPageAC(1)
        let newState = postsReducer(state, action)

        expect(newState.currentPage).toBe(1)
    })
    it("currentPage should be two", ()=>{
        let action = setCurrentPageAC(2)
        let newState = postsReducer(state, action)

        expect(newState.currentPage).toBe(2)
    })
})

describe("------- SET TOTAL POSTS COUNT ------", ()=>{
    it("totalPostsCount should be three", ()=>{
        let action = setTotalPostsCountAC(3)
        let newState = postsReducer(state, action)

        expect(newState.totalPostsCount).toEqual(3)
    })
})
