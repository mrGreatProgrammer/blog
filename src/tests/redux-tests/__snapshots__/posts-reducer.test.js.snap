// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---- GET ALL POSTS ---- posts values should be correct 1`] = `
Array [
  Object {
    "id": 1,
    "info": undefined,
    "isModalVisible": false,
    "liked": false,
    "likes": 0,
    "textContent": undefined,
    "title": "first post title",
    "userId": 1,
  },
  Object {
    "id": 2,
    "info": undefined,
    "isModalVisible": false,
    "liked": false,
    "likes": 0,
    "textContent": undefined,
    "title": "second post title",
    "userId": 3,
  },
  Object {
    "id": 3,
    "info": undefined,
    "isModalVisible": false,
    "liked": false,
    "likes": 0,
    "textContent": undefined,
    "title": "third post title",
    "userId": 1,
  },
]
`;
